primitives/src/math/fixed.rs

```diff
@@ -252,21 +256,27 @@ impl<F: Fixed + ToString, N: TryFrom<u128>> FromFixedToDecimal<F> for N {
         }
 
         let mut increment_int_part = false;
-        let new_frac_part = if frac_part.len() < decimals_usize {
-            format!("{}{}", frac_part, "0".repeat(decimals_usize.saturating_sub(frac_part.len())))
-        } else if frac_part.len() > decimals_usize {
-            let round_digit =
-                frac_part.chars().nth(decimals_usize).and_then(|c| c.to_digit(10)).ok_or(
-                    "The char at decimals_usize was not found, although the frac_part length is \
-                     higher than decimals_usize.",
-                )?;
-            if round_digit >= 5 {
-                increment_int_part = true;
+
+        let new_frac_part = match frac_part.len().cmp(&decimals_usize) {
+            Ordering::Less => {
+                format!(
+                    "{}{}",
+                    frac_part,
+                    "0".repeat(decimals_usize.saturating_sub(frac_part.len()))
+                )
             }
-            frac_part.chars().take(decimals_usize).collect::<String>()
-        } else {
-            // There's no need to round here, the fractional part already fits perfectly.
-            frac_part.to_string()
+            Ordering::Greater => {
+                let round_digit =
+                    frac_part.chars().nth(decimals_usize).and_then(|c| c.to_digit(10)).ok_or(
+                        "The char at decimals_usize was not found, although the frac_part length \
+                         is higher than decimals_usize.",
+                    )?;
+                if round_digit >= 5 {
+                    increment_int_part = true;
+                }
+                frac_part.chars().take(decimals_usize).collect::<String>()
+            }
+            Ordering::Equal => frac_part.to_string(),
```

Comment:
Looking at this now, this arm is just a special case of the first arm. You could just do the `format!` call even if `frac_part` has length `decimals_usize`.